tests/test_appsettings.py

```diff
@@ -560,6 +560,52 @@ def test_parent_setting_precedence_over_environ_value(self):
         assert setting.value["a"] == "A"
         assert setting.value["b"] == "B"
 
+    def test_nested_dict_nested_dict(self):
+        setting = appsettings.NestedDictSetting(
+            settings=dict(
+                sub=appsettings.NestedDictSetting(settings=dict(value=appsettings.Setting(name="value")), name="sub")
+            ),
+            name="setting",
+        )
+
+        setting.check()
+        assert setting.value == {}
+
+        with override_settings(SETTING={}):
+            setting.check()
+            assert setting.value["sub"] == {}
+
+        with override_settings(SETTING={"SUB": {}}):
+            setting.check()
+            assert setting.value["sub"]["value"] is None
+
+        with override_settings(SETTING={"SUB": {"VALUE": "Value"}}):
+            setting.check()
+            assert setting.value["sub"]["value"] == "Value"
+
+    def test_nested_dict_nested_list(self):
+        setting = appsettings.NestedDictSetting(
+            settings=dict(
+                sub=appsettings.NestedListSetting(
+                    inner_setting=appsettings.Setting(name="value", default=None), name="sub", default=None
+                )
+            ),
+            name="setting",
+            default=None,
+        )
+
+        setting.check()
+        assert setting.value is None
+
+        with override_settings(SETTING={}):
+            setting.check()
+            assert setting.value is not None
+            assert setting.value["sub"] is None
```

Comment:
I believe `assert setting.value == {'sub': None}` would be stricter and more descriptive. Similar construction could be also used for other asserts.

Reply:
Fixed.

Reply:
The `assert setting.value is not None` is still there, although it duplicates the following assert.

Reply:
Sorry, forgot to fix this. Now it should be done.